glint/app/app.odin

```
@@ -0,0 +1,120 @@
+package app
+
+import "core:c"
+import sg "sokol:gfx"
+import slog "sokol:log"
+import "vendor:glfw"
+
+SAMPLE_COUNT: c.int : 1
+
+Glint_App_Err :: enum {
+	Glfw_Init_Failed,
+	Glfw_Window_Failed,
+}
+
+Glint_App :: struct {
+	state: Glfw_State,
+	desc:  Desc,
+}
+
+Glint_App_Result :: union {
+	Glint_App,
+	Glint_App_Err,
+}
+
+init :: proc(desc: Desc) -> Glint_App_Result {
```

Please use multiple return values instead of a union:
```odin
init :: proc(desc: Desc) -> (Glint_App, Glint_App_Err) {
```
On error, you can just return a default constructed Glint_App:
```odin
return {}, .Glfw_Init_Failed
```

Also, you could return the error string from glfw. I am pretty sure that there is a function for that in glfw.

Also, please don't call this function `init` but either something like `create` or `make`. Use `init` for functions that initialise an already existing `Glint_App` that is passed via a pointer.